verticapy/tests/vModel/test_svd.py

```diff
@@ -0,0 +1,165 @@
+# limitations under the License.
+
+import pytest, warnings, sys
+from verticapy.learn.decomposition import SVD
+from verticapy import drop_table
+
+from verticapy import set_option
+
+set_option("print_info", False)
+
+
+@pytest.fixture(scope="module")
+def winequality_vd(base):
+    from verticapy.learn.datasets import load_winequality
+
+    winequality = load_winequality(cursor=base.cursor)
+    yield winequality
+    with warnings.catch_warnings(record=True) as w:
+        drop_table(name="public.winequality", cursor=base.cursor)
+
+
+@pytest.fixture(scope="module")
+def model(base, winequality_vd):
+    base.cursor.execute("DROP MODEL IF EXISTS SVD_model_test")
+    model_class = SVD("SVD_model_test", cursor=base.cursor)
+    model_class.fit(
+        "public.winequality", ["citric_acid", "residual_sugar", "alcohol"]
+    )
+    yield model_class
+    model_class.drop()
+
+class TestSVD:
+
+    def test_deploySQL(self, model):
+        expected_sql = 'APPLY_SVD("citric_acid", "residual_sugar", "alcohol" USING PARAMETERS model_name = \'SVD_model_test\', match_by_pos = \'true\', cutoff = 1)'
+        result_sql = model.deploySQL()
+
+        assert result_sql == expected_sql
+
+    def test_drop(self, base):
+        base.cursor.execute("DROP MODEL IF EXISTS SVD_model_test_drop")
+        model_test = SVD("SVD_model_test_drop", cursor=base.cursor)
+        model_test.fit("public.winequality", ["alcohol", "quality"])
+
+        base.cursor.execute(
+            "SELECT model_name FROM models WHERE model_name = 'SVD_model_test_drop'"
+        )
+        assert base.cursor.fetchone()[0] == "SVD_model_test_drop"
+
+        model_test.drop()
+        base.cursor.execute(
+            "SELECT model_name FROM models WHERE model_name = 'SVD_model_test_drop'"
+        )
+        assert base.cursor.fetchone() is None
+
+    def test_get_attr(self, model):
+        m_att = model.get_attr()
+
+        assert m_att["attr_name"] == [
+            "columns",
+            "singular_values",
+            "right_singular_vectors",
+            "counters",
+            "call_string",
+        ]
+        assert m_att["attr_fields"] == [
+            "index, name",
+            "index, value, explained_variance, accumulated_explained_variance",
+            "index, vector1, vector2, vector3",
+            "counter_name, counter_value",
+            "call_string",
+        ]
+        assert m_att["#_of_rows"] == [3, 3, 3, 3, 1]
+
+        m_att_details = model.get_attr(attr_name="singular_values")
+
+        assert m_att_details["value"][0] == pytest.approx(968.964362586858, abs=1e-6)
+        assert m_att_details["value"][1] == pytest.approx(354.585184720344, abs=1e-6)
+        assert m_att_details["value"][2] == pytest.approx(11.7281921567471, abs=1e-6)
+
+    def test_get_params(self, model):
+        assert model.get_params() == {'method': 'lapack', 'n_components': 0}
+
+    def test_to_sklearn(self, model):
+        # Zscore
```

Comment:
The `# Zscore` comment looks irrelevant.

Reply:
Erased!